docs/troubleshooting.md

```diff
@@ -0,0 +1,341 @@
+
+### Timed out while failing to retrieve kubeconfig secret
+
+When `clusterctl` times out waiting for the management cluster to come online, and the vSphere manager log repeats `failed to retrieve kubeconfig secret for Cluster` over and over again, it means there was an error bringing the management cluster's first control plane node online. Possible resaons include:
+
+#### Cannot access the vSphere endpoint
+
+Two common causes for a failed deployment are related to accessing the remote vSphere endpoint:
+
+1. The host from which `clusterctl` is executed must have access to the vSphere endpoint to which the management cluster is being deployed.
+2. The provided vSphere credentials are invalid.
+
+A quick way to validate both access and the credentials is using the program [`govc`](https://github.com/vmware/govmomi/tree/master/govc) or its container image, [`vmware/govc`](https://hub.docker.com/r/vmware/govc):
+
+```shell
+# Define the vSphere's endpoint and access information.
+$ export GOVC_URL="myvcenter.com" GOVC_USERNAME="username" GOVC_PASSWORD="password"
+
+# Use "govc" to list the contents of the vSphere endpoint.
+$ docker run --rm \
+  -e GOVC_URL -e GOVC_USERNAME -e GOVC_PASSWORD \
+  vmware/govc \
+  ls -k
+/my-datacenter/vm
+/my-datacenter/network
+/my-datacenter/host
+/my-datacenter/datastore
+```
+
+If the above command fails then there is an issue with accessing the vSphere endpoint, and it must be corrected before `clusterctl` will succeed.
+
+#### A VM with the same name already exists
+
+Deployed VMs get their names from the names of the machines in `machines.yaml` and `machineset.yaml`. If a VM with the same name already exists in the same location as one of the VMs that would be created by a new cluster, then the new cluster will fail to deploy and the CAPV manager log will include an error similar to the following:
+
+```shell
+I0726 18:52:48.920975       1 util.go:195] default-logger/cluster.k8s.io/v1alpha1/default/v0.4.0-beta.2/management-cluster-controlplane-1/task-231288 "level"=2 "msg"="task failed"  "description-id"="VirtualMachine.clone"
+```
+
+Use the `govc` image to check to see if there is a VM with the same name:
+
+```shell
+$ docker run --rm \
+  -e GOVC_URL -e GOVC_USERNAME -e GOVC_PASSWORD \
+  vmware/govc \
+  vm.info -k management-cluster-controlplane-1
+Name:           management-cluster-controlplane-1
+  Path:         /my-datacenter/vm/management-cluster-controlplane-1
+  UUID:         4230a650-c92a-d99a-d9f7-fa2fd770e536
+  Guest name:   Other 3.x or later Linux (64-bit)
+  Memory:       2048MB
+  CPU:          2 vCPU(s)
+  Power state:  poweredOn
+  Boot time:    <nil>
+  IP address:   5.6.7.8
+  Host:         1.2.3.4
+```
+
+If the above VM is no longer needed then it may be deleted using `govc` as well:
+
+```shell
+docker run --rm \
+  -e GOVC_URL -e GOVC_USERNAME -e GOVC_PASSWORD \
+  vmware/govc \
+  vm.destroy -k management-cluster-controlplane-1
```

Comment:
Is it safe to recommend this without setting DC, ResourcePool, etc?

Reply:
Fair.